portal-frontend/src/app/features/applications/edit-submission/primary-contact/primary-contact.component.ts

```diff
@@ -275,4 +287,70 @@ export class PrimaryContactComponent extends FilesStepComponent implements OnIni
   private prepareGovernmentOwners() {
     this.parcelOwners = [];
   }
+
+  async onSelectPrimaryContactType(isExistingOwner: boolean | null) {
+    if (this.form.dirty || (this.selectedOwnerUuid !== 'agent' && this.selectedOwnerUuid !== 'government')) {
+      await this.dialog
+        .open(PrimaryContactConfirmationDialogComponent, {
+          panelClass: 'no-padding',
+          disableClose: true,
+        })
+        .beforeClosed()
+        .subscribe(async (confirmed) => {
+          if (confirmed) {
+            this.switchPrimaryContactType(isExistingOwner);
+          } else {
+            this.isExistingOwner.setValue(!isExistingOwner);
+          }
+        });
+    } else {
+      this.switchPrimaryContactType(isExistingOwner);
+    }
+  }
+
+  switchPrimaryContactType(isExistingOwner: boolean | null) {
+    if (!isExistingOwner) {
+      this.onSelectAgent();
+    } else if (this.parcelOwners.length == 1) {
+      this.onSelectOwner(this.parcelOwners[0].uuid);
+    } else if (this.isGovernmentUser) {
+      this.onSelectGovernment();
+    } else {
+      this.selectedThirdPartyAgent = false;
+    }
```

Comment:
If a government user is submitting an applicant on behalf of a single parcel owner and they select 'Yes' to the toggle question, based on this logic, the parcel owner will be the primary contact instead of the expected result of the government employee.

Also use strict equality `===` instead